### PR TITLE
nixos/prometheus-bitcoin-exporter: init

### PR DESCRIPTION
**Pull request by mmilata**
* ~~Currently a draft because it depends on #89247~~
* Adds `riprova` python module as a dependency.
* ~~The `listenAddress` module option currently does nothing, PR: jvstein/bitcoin-prometheus-exporter#11.~~ (merged to master, included as a patch)

###### Motivation for this change

I like to monitor things.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by mdlayher — COMMENTED** [review]
I don't know much about the Python bits but left some comments on the Prometheus aspects.

**Comment by mmilata** [comment]
#89247 has been merged & this is ready for review, PTAL.

**Comment by mmilata** [comment]
Rebased to master & fixed test failure due to port number collision.

**Comment by prusnak** [comment]
@risicle do you want to have a look at this one too?

File: pkgs/development/python-modules/riprova/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "04drdvjjbh370csv2vb5zamg2aanxqkfm6w361qkybnra4g4g0dz";
+  };
+
+  propagatedBuildInputs = [ six ];
+
+  # PyPI archive doesn't have tests
+  doCheck = false;
+
```
Comment (prusnak) [inline pkgs/development/python-modules/riprova/default.nix:20]: We can at least do:

```python
pythonImportsCheck = [ "riprova" ];
```

**Comment by risicle** [comment]
I don't really know anything about bitcoin.

File: pkgs/development/python-modules/riprova/default.nix (hunk 1 above)
Comment (mmilata) [reply pkgs/development/python-modules/riprova/default.nix:20]: Fixed, totally forgot that it exists.

**Comment by prusnak** [comment]
@GrahamcOfBorg build python3Packages.riprova prometheus-bitcoin-exporter

**Pull request merged**